### ISSUE TITLE
THANK YOU FOR THIS GREAT CHRISTMAS PRESENT TOMAS!!! Happy Christmas and New Year!

### ISSUE THREAD

**Issue by GuerreroNinja** [MiSTer-devel/MSX1_MiSTer#4]
Between the presents I received today, I found one gem that made me jump of joy: a MSX2 core test!!! Wuuuut!!

BEST PRESENT EVER!

Thank you Tomas! Enjoy the festivities and Happy New Year!

**Comment by nonochus** [comment]
Hello Tomas, thank you very much for the msx2 core test, as guerreroninja says, it is a great start and there are still things to do, but I was also very excited to try it.

I have dumped my own ROMS from my msx 2, a philips vg8235 but they are  modified roms, they should boot at 60hz and if you press the "h" key in the boot logo, it should change to 50hz, this modification was done because the european msx boot at 50hz and play certain games on cartridge at 50hz it lost a lot, the extra 10hz of ntsc is very noticeable, also, the ram counter on the start screen with the msx logo is working, it should tell you how much ram that the computer has.if you have 256k it tells you in boot logo. I attach them in case they help you to improve the core and know how a msx 2 behaves in real hardware, just to test the core and improve it if you want to take a look maybe helps.

Thank you very much for the core, I will be happy to try it and help in any way I can to improve it and MERRY CHRISTMAS!
[MSX2 BIOS.zip](https://github.com/MiSTer-devel/MSX1_MiSTer/files/10300586/MSX2.BIOS.zip)

**Comment by nonochus** [comment]
Tried the new core test, the video modes with my custom "roms" work ok, as they should. Tested on a crt tv at 15khz.

The ram seems to still not work, it always marks 4mb whatever you choose in ram size, I guess you'll still be working on it.

Please, tell me if I can help you with something or if you prefer to work quietly and peacefully without anyone bothering you.

I could tell you what things I'm trying and little by little improve it or if you prefer, I won't say anything else.

Very thankful!

**Comment by Vampier** [comment]
Yes that was quite a pleasant surprise - I am writing test programs which I am sharing on the #MSX channel in the official MiSTer discord server https://discord.gg/misterfpga (you can find us in MiSTer FPGA > MiSTer Cores > MSX or just message me if you're on the server)
